README.md

```diff
@@ -2,19 +2,134 @@
 
+### Enable engine integration with 3rd parties
+
+- [ ] gitbase-spark-connector SQL 3rd party interop (Spark Thrift Server) [P0][DP]
+
+### Increase number of algorithms we can run
+
+- [ ] Support for diff tree [P0][DP]
+- [ ] Support for diff [P0][DP]
+- [ ] Research options for UDFs (gitbase or gitbase-spark-connector) [P1][DP]
+- [ ] Research options for gitbase plugins [P2][DP]
+- [ ] Gitbase support for low-level git UDFs [P2][DP]
+
+### Increase variety of data we can analyze
+
+- [ ] Provide specs for external data sources [P1][DP][PRODUCT]
+- [ ] Provide specs to expose dependency analysis [P2][LA]
+
+### Analyze current performance limits and improve them
+
+- [ ] Gitbase to expose only reachable objects [P0][DP]
+- [ ] Parallelized index creation [P1][DP]
+- [ ] Parallelized queries inside of a repository [P2][DP]
+- [ ] Make performance dashboard available to product team [P1][DP][PRODUCT]
+- [ ] Provide specs for distributed version (gitbase-spark-connector-enterprise) [P2][DP][INFRA]
+
+### Make enterprise friendly
+
+- [ ] gitbase-web SSO and RBAC [P1][APPS][PRODUCT]
+- [ ] bblfsh-web SSO and RBAC [P1][APPS][PRODUCT]
+- [ ] document gitbase audit logs and RBAC [P1][DP]
+
+### Simplify usage
+
+- [ ] Provide installers [P2][APPS][PRODUCT]
+- [ ] Provide specs for new Engine architecture (stand-alone server + config file) [PRODUCT]
+
+### Improve stability
+
+- [ ] Ensure gitbase works when dataset and indexes do not match [P0][DP]
+- [ ] Provide a single, default, performant and stable regex implementation for gitbase and enry [P2][DP][LA]
+- [ ] Stabilize Babelfish in production (support large-scale use from gitbase) [P0][LA]
+- [ ] Add engine cli integration tests [P0][APPS]
+
+### Online updates
+
+- [ ] Gitbase to support online updates [P1][DP]
+
+### Release necessary language support for enterprise
+
+- [ ] C# driver [P0][LA]
+
+### Get bblfsh ready for wide language support of our use cases
+
+- [ ] Make UAST v2 the standard interface for Babelfish [P0][LA]
+- [ ] Compact diffs of UAST trees [P1][LA]
+- [ ] Plan for a language-independent style-analyzer (bblfsh-side support) [P2][LA]
+- [ ] Provide an efficient storage format for large UASTs collections [P3][LA]
+
 ## Improve the 'Code as Data' individual user experience
 
 ## Improve quality across all products
 
+- [ ] Conduct two empathy sessions for cross-team documentation and user experience improvements [P1][ENG]
+
 ## Release a new version of PGA
 
+- [ ] Release a new version, updated, fixed, including #stars [P1][DR]
+
 ## Bring 'Assisted Code Review' to developers in the open-source community
 
+### Improve current lookout analyzers
+
+- Style Analyzer in one more language (possibly Python) [P0][ML]
+- Release Typo Analyzer [P0][ML]
+
+### Create new lookout analyzers
+
+- [ ] Analyzers for most common linters / traditional static tools [P?][APPS][PRODUCT]
+- [ ] Collect a structured dataset of diffs for best-practices-analyzer [P1][ML]
```

Comment:
```suggestion
- [ ] Collect a structured dataset of diffs for best-practices-analyzer [P2][ML]
```

since is blocked on the UAST diff support in Babelfish